src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/MinMax_PowerFxV1CompatibilityRules.txt

```diff
@@ -0,0 +1,346 @@
+﻿#SETUP: PowerFxV1CompatibilityRules
+
+// With this flag enabled, return type is
+// - always argTypes[0] if all elements are [Date, Time, DateTime, Number, Decimal]
+// - otherwise it's Decimal or Number, depending on NumberIsFloat flag
+
+>> Max(1)
+1
+
+>> Min(1)
+1
+
+>> Max(1,2)
+2
+
+>> Min(1,2)
+1
+
+>> Max(0)
+0
+
+>> Min(0)
+0
+
+>> Max(0,0,0,0)
+0
+
+>> Min(0,0,0,0)
+0
+
+>> Max(1, 2, 3, 4, 5)
+5
+
+>> Min(1, 2, 3, 4, 5)
+1
+
+>> Max(-1, -2, -3, -4, -5)
+-1
+
+>> Min(-1, -2, -3, -4, -5)
+-5
+
+>> Max(2.34567)
+2.34567
+
+>> Min(2.34567)
+2.34567
+
+>> Max(-1.23, 23.4, -345.6, 4567.89, -5678.912)
+4567.89
+
+>> Min(-1.23, 23.4, -345.6, 4567.89, -5678.912)
+-5678.912
+
+>> Max(-123, 234, -3456, 4567, -5678)
+4567
+
+>> Min(-123, 234, -3456, 4567, -5678)
+-5678
+
+>> Max(-100, If(true, -1000, 200))
+-100
+
+>> Min(-100, If(true, -1000, 200))
+-1000
+
+>> Max(Blank())
+Blank()
+
+>> Min(Blank())
+Blank()
+
+>> Max(Blank(), Blank(), Blank())
+Blank()
+
+>> Min(Blank(), Blank(), Blank())
+Blank()
+
+>> Min(1, 2, 3, Blank(), 2, 3)
+1
+
+>> Max(1, 2, 3, Blank(), 2, 3)
+3
+
+>> Min(-1, -2, -3, Blank(), -2, -3)
+-3
+
+>> Max(-1, -2, -3, Blank(), -2, -3)
+-1
+
+// ****** Date/time arguments, scalar arguments ******
+
+>> Max(Min(Date(2022,3,8)))
+Date(2022,3,8)
+
+>> Max(Date(1985,3,5),Date(1986,10,19))
+Date(1986,10,19)
+
+>> Min(Date(1985,3,5),Date(1986,10,19))
+Date(1985,3,5)
+
+>> Max(DateTime(2014,11,10,13,50,0), DateTime(2014,11,11,13,50,0))
+DateTime(2014,11,11,13,50,0,0)
+
+>> Min(DateTime(2014,11,10,13,50,0), DateTime(2014,11,11,13,50,0))
+DateTime(2014,11,10,13,50,0,0)
+
+>> Max(DateTime(2014,11,10,13,50,0), DateTime(2014,11,10,14,50,0))
+DateTime(2014,11,10,14,50,0,0)
+
+>> Min(DateTime(2014,11,10,13,50,0), DateTime(2014,11,10,14,50,0))
+DateTime(2014,11,10,13,50,0,0)
+
+>> Max(Time(1,50,0),Time(2,0,0))
+Time(2,0,0,0)
+
+>> Min(Time(1,50,0),Time(2,0,0))
+Time(1,50,0,0)
+
+>> Max(Date(2022,3,8),DateTime(2022,3,8,13,50,0))
+Date(2022,3,8)
+
+>> Min(Date(2022,3,8),DateTime(2022,3,8,13,50,0))
+Date(2022,3,8)
+
+// ****** Mix and numeric and date/time arguments, scalar arguments (result is numeric) ******
+
+// Date(2022, 9, 9) = 44813
+>> Max(Date(2022, 9, 9), 50000)
+Date(2036,11,21)
+
+>> Min(Date(2022, 9, 9), 50000)
+Date(2022,9,9)
+
+>> Max(Date(2022, 9, 9), 40000)
+Date(2022,9,9)
+
+>> Min(Date(2022, 9, 9), 40000)
+Date(2009,7,6)
+
```

Comment:
It would be good to have the same test with the arguments switched, to show that the result will be a number